pkg/common/cns-lib/volume/manager.go

```diff
@@ -2112,6 +2112,56 @@ func (m *defaultManager) deleteSnapshotWithImprovedIdempotencyCheck(
 	// Get the taskInfo
 	deleteSnapshotsTaskInfo, err := cns.GetTaskInfo(ctx, deleteSnapshotTask)
 	if err != nil || deleteSnapshotsTaskInfo == nil {
+		if err != nil && cnsvsphere.IsManagedObjectNotFound(err, deleteSnapshotTask.Reference()) {
+			log.Infof("Snapshot %q on volume %q might have already been deleted "+
+				"with the error %v. Calling CNS QuerySnapshots API to confirm it", snapshotID, volumeID,
+				spew.Sdump(err))
+
+			// To double check whether the DeleteSnapshot task is completed in the underlying vSphere.
+			snapshotQueryFilter := cnstypes.CnsSnapshotQueryFilter{
+				SnapshotQuerySpecs: []cnstypes.CnsSnapshotQuerySpec{
+					{VolumeId: cnstypes.CnsVolumeId{Id: volumeID}, SnapshotId: &cnstypes.CnsSnapshotId{Id: snapshotID}},
+				},
+				Cursor: &cnstypes.CnsCursor{Offset: 0, Limit: 1},
+			}
```

Comment:
Not sure if it's ideal in this case, but https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/fcd6dae8135571e24bec74856bf8e8f50d649fca/pkg/csi/service/common/vsphereutil.go#L683 method does this.